docs/adrs/0002-helm-charts-as-profiles.md

```diff
@@ -0,0 +1,52 @@
+# 2. Helm Charts as Profiles
+
+Date: 2021-10-20
+
+## Status
+
+Accepted
+
+## Context
+
+We want to build a set of packages for Kubernetes, that we can use as building
+blocks for configuring clusters.
+
+These are versioned, and can provide simplified configuration for these
+packages, where they might incorporate sub-packages.
+
+## Decision
+
+Profiles are Helm charts.
+
+They can be Helm charts with [Subcharts](https://helm.sh/docs/chart_template_guide/subcharts_and_globals/) or [dependencies](https://helm.sh/docs/chart_best_practices/dependencies/#helm).
+
+There's a conceptual difference between Profiles and Helm charts, but the
+technical implementation is implemented entirely in Helm charts.
+
+To identify whether or not a Helm chart is a Profile or not:
+
+This proposes `weave.works/profile: name-of-profile` as the annotation, for
+example `Chart.yaml`.
+
+```yaml
+name: demo-profile
+version: 0.0.1
+annotations:
+  weave.works/profile: "A Demo Profile"
```

Comment:
Hm, I thought this will be just a `true/false` thing. Is this defining the name of the profile? Is that redundant information or will this be definitive, or just used as a description? Are we supposed to use this value for something is what I'm asking. :)